src/inmanta/agent/executor.py

```diff
@@ -311,6 +314,21 @@ def get_or_create_env_directory(self, blueprint: EnvBlueprint) -> tuple[str, boo
             )
             return env_dir, False  # Returning the path and False for existing directory
 
+    def read_environment_status(self, inmanta_env_status_file: str) -> Optional[str]:
```

Comment:
style wise, I would make this method take a folder as argument and have it return a VirtualEnvStatus

```suggestion
    def read_environment_status(self, venv_folder: str) -> Optional[VirtualEnvStatus]:
```

i.e. I would make this method encapsulate the manner in which this value is persisted. 

And I would make an associated write method

(you could even group these methods on a class, or add them to the `ExecutorVirtualEnvironment` but I'm less opinionated about that)